common/replication/block_fetcher.go

```diff
@@ -509,9 +527,14 @@ func (bf *BlockFetcher) PullBlock(seq uint64) *common.Block {
 
 		if block != nil {
 			bf.Logger.Debugf("Got block %d from %s", block.Header.Number, bf.currentEndpoint.Endpoint)
-			if err := bf.VerifyBlock(block.Header, block.Metadata); err != nil {
-				bf.Logger.Warnf("Failed verifying signature on block %d received from %s: %v", block.Header.Number, bf.currentEndpoint.Endpoint, err)
-				return nil
+			/* In case `VerifyBlock` Function is not set, genesis block is being pulled, so no need to verify it.*/
```

Comment:
it's not that there is "no need to verify it". You cannot verify it since it's not signed by an orderer. 

But you need to pull it from different sources and compare it.

Reply:
Why shuffle? Just pull from everyone you can, then compare. If you get `f+1` identical matches, this is the genesis block to use.